File: NetFramework/FlexGrid/VB/RowDetails/Samples/CustomSample.vb
Imports C1.Win.C1FlexGrid

Public Class CustomSample
    Private Sub CustomSample_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim conn As String = Util.GetConnectionString()
        Dim ds = New DataSet()
        Util.FillTable(ds, "Employees", conn)

        flexGrid.DataSource = ds
        flexGrid.DataMember = "Employees"

        flexGrid.RowDetailProvider = Function(g, r) New CustomRowDetail()
        flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected

        flexGrid.Cols("Notes").Visible = False
    End Sub
End Class

File: NetFramework/FlexGrid/CS/RowDetails/Samples/CustomSample.cs
using System;
using System.Data;
using System.Windows.Forms;
using C1.Win.C1FlexGrid;

namespace RowDetails
{
    public partial class CustomSample : UserControl
    {
        public CustomSample()
        {
            InitializeComponent();
        }

        private void CustomSample_Load(object sender, EventArgs e)
        {
            if (DesignMode) return;

            string conn = Util.GetConnectionString();
            var ds = new DataSet();
            Util.FillTable(ds, "Employees", conn);

            flexGrid.DataSource = ds;
            flexGrid.DataMember = "Employees";
            flexGrid.RowDetailProvider = (g, r) => new CustomRowDetail();
            flexGrid.RowDetailsVisibilityMode = RowDetailsVisibilityMode.VisibleWhenSelected;
            flexGrid.Cols["Notes"].Visible = false;
        }
    }
}
